mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -238,7 +245,10 @@ def evaluate_dictstatement(self, cur: mparser.DictNode) -> InterpreterObject:
         def resolve_key(key: mparser.BaseNode) -> str:
             if not isinstance(key, mparser.StringNode):
                 FeatureNew.single_use('Dictionary entry using non literal key', '0.53.0', self.subproject)
-            str_key = _unholder(self.evaluate_statement(key))
+            key_holder = self.evaluate_statement(key)
+            if key_holder is None:
+                raise InvalidCodeOnVoid()
```

Comment:
Catches:

```meson
d = {add_global_arguments('-arg', language: 'c') : 'bar'}
```
Values are already caught in reduce arguments as "Value of key bar is void".